Common/Mapping/MapDeviceInterface.cs

```diff
@@ -153,6 +153,54 @@ public static void Close(bool fromEntity = false)
 	}
 }
 
+internal class MapDeviceShiftClickPlayer : ModPlayer
+{
+	public override bool HoverSlot(Item[] inventory, int context, int slot)
+	{
+		if (context != ItemSlot.Context.InventoryItem)
+		{
+			return false;
+		}
+
+		if (SmartUiLoader.TryGetUiState(out MapDeviceState? map) && map is { Visible: true } && MapDeviceInterface.Entity is not null && ItemSlot.ShiftInUse)
+		{
+			Main.cursorOverride = CursorOverrideID.InventoryToChest;
+			return true;
+		}
+		
+		return false;
+	}
+
+	public override bool ShiftClickSlot(Item[] inventory, int context, int slot)
+	{
+		if (context != ItemSlot.Context.InventoryItem)
+		{
+			return false;
+		}
+
+		if (SmartUiLoader.TryGetUiState(out MapDeviceState? map) && map is { Visible: true } && MapDeviceInterface.Entity is not null)
+		{
+			Item[] storage = MapDeviceInterface.Entity.Storage;
+
+			for (int i = 0; i < MapDeviceEntity.StorageSize; ++i)
+			{
+				ref Item item = ref storage[i];
+
+				if (item.IsAir)
+				{
+					Item invItem = inventory[slot].Clone();
+					inventory[slot].TurnToAir();
+					item = invItem;
+				}
+			}
+
+			return true;
+		}
+
+		return false;
```

Comment:
Break the loop, play audio. Not entirely sure whether true should be returned only on success, or always if the UI is open.
```suggestion
			for (int i = 0; i < MapDeviceEntity.StorageSize; i++)
			{
				ref Item item = ref storage[i];
				if (!item.IsAir) { continue; }

				Item invItem = inventory[slot].Clone();
				inventory[slot].TurnToAir();
				item = invItem;
				SoundEngine.PlaySound(SoundID.MenuTick);
				break;
			}

			return true;
		}

		return false;
```